### ISSUE TITLE
Server not starting

### ISSUE THREAD

**Issue by SBRyno** [azixus/ARK_Ascended_Docker#9]
I've been trying to get the ASA server up on my proxmox machine that has Ubuntu LTS running on a virtual machine. With ASE, i was able to run ark servers using LGSM. For ASA LGSM is not an option so i have tried your Docker solution. I followed your instructions, edited the .env, and oped port 7790 in my router. Looking at the log, it doesn't start fully. Here are the relevant log entries:
`asa_server  | -------- STARTING SERVER --------
asa_server  | tail: /opt/arkserver/ShooterGame/Saved/Logs/ShooterGame.log: file truncated
asa_server  | [2023.11.05-03.37.02:238][  0]Log file open, 11/05/23 03:37:02
asa_server  | [2023.11.05-03.37.02:239][  0]LogMemory: Platform Memory Stats for WindowsServer
asa_server  | [2023.11.05-03.37.02:239][  0]LogMemory: Process Physical Memory: 331.25 MB used, 335.71 MB peak
asa_server  | [2023.11.05-03.37.02:239][  0]LogMemory: Process Virtual Memory: 302.62 MB used, 302.62 MB peak
asa_server  | [2023.11.05-03.37.02:239][  0]LogMemory: Physical Memory: 944.05 MB used,  31149.34 MB free, 32093.39 MB total
asa_server  | [2023.11.05-03.37.02:239][  0]LogMemory: Virtual Memory: 944.05 MB used,  39341.34 MB free, 40285.39 MB total
asa_server  | [2023.11.05-03.37.02:950][  0]ARK Version: 26.2
asa_server  | [2023.11.05-03.37.04:160][  8]Primal Game Data Took 0.00 seconds
asa_server  | [2023.11.05-03.37.28:873][  8]LogMemory: Platform Memory Stats for WindowsServer
asa_server  | [2023.11.05-03.37.28:873][  8]LogMemory: Process Physical Memory: 636.85 MB used, 649.10 MB peak
asa_server  | [2023.11.05-03.37.28:873][  8]LogMemory: Process Virtual Memory: 589.93 MB used, 589.93 MB peak
asa_server  | [2023.11.05-03.37.28:873][  8]LogMemory: Physical Memory: 1438.58 MB used,  30654.82 MB free, 32093.39 MB total
asa_server  | [2023.11.05-03.37.28:873][  8]LogMemory: Virtual Memory: 1438.58 MB used,  38846.81 MB free, 40285.39 MB total
asa_server  | -------- STARTING SERVER --------
asa_server  | -------- STARTING SERVER --------
asa_server  | -------- STARTING SERVER --------`

**Comment by HylianSeven** [comment]
I'm currently having the same issue as well, but running Debian on my actual machine and not a VM. I'm seeing the same thing you are in the logs.

**Comment by azixus** [comment]
In my experience, when the platform memory stats appear twice, it's usually one of those issues:
- Corrupted .arktribe files (this usually happens after playing for a while, current ASA bug). You can fix it by removing the arktribe that is bigger than the rest.
- Not enough memory. You need at least 12GB free memory for the server to start, not the problem of @SBRyno.
- Incoherent configuration between the server command line and GUS.ini (e.g. different session names). Try to do a `docker compose down`, `docker compose up --build -d`.
- Bad modid, e.g. using the id in the URL instead of the Project ID.
- The .env file is not parsed properly by docker. Try to remove quotes inside the .env file. This should be fixed in the dev branch, will merge soon.

If none of these appear to be your problem, dm me on Discord, username is the same.

@EvilSeeQ, I just tested starting the server with two mods and it appear to work fine for me:
```
asa_server2  | tail: /opt/arkserver/ShooterGame/Saved/Logs/ShooterGame.log: file truncated
asa_server2  | ﻿[2023.11.05-09.02.15:926][  0]Log file open, 11/05/23 09:02:15
asa_server2  | [2023.11.05-09.02.15:941][  0]LogMemory: Platform Memory Stats for WindowsServer
asa_server2  | [2023.11.05-09.02.15:941][  0]LogMemory: Process Physical Memory: 330.38 MB used, 333.79 MB peak
asa_server2  | [2023.11.05-09.02.15:941][  0]LogMemory: Process Virtual Memory: 301.92 MB used, 301.92 MB peak
asa_server2  | [2023.11.05-09.02.15:941][  0]LogMemory: Physical Memory: 32649.93 MB used,  31520.36 MB free, 64170.30 MB total
asa_server2  | [2023.11.05-09.02.15:941][  0]LogMemory: Virtual Memory: 46379.02 MB used,  50526.28 MB free, 96905.30 MB total
asa_server2  | [2023.11.05-09.02.17:686][  0]ARK Version: 26.3
asa_server2  | [2023.11.05-09.02.21:777][ 55]Primal Game Data Took 0.00 seconds
asa_server2  | [2023.11.05-09.03.19:399][ 55]Server: "My Awesome ASA Server" has successfully started!
asa_server2  | [2023.11.05-09.03.20:749][ 55]Commandline:  TheIsland_WP?listen?SessionName="My Awesome ASA Server"?Port=7791?MaxPlayers=10?ServerPassword=MyServerPassword?ServerAdminPassword="MyArkAdminPassword"?RCONEnabled=True?RCONPort=32330?ServerCrosshair=true?OverrideStructurePlatformPrevention=true?OverrideOfficialDifficulty=5.0?ShowFloatingDamageText=true?AllowFlyerCarryPvE=true -mods=929420,928793 -log -NoBattlEye -WinLiveMaxPlayers=10 -ForceAllowCaveFlyers -ForceRespawnDinos -AllowRaidDinoFeeding=true -ActiveEvent=Summer
asa_server2  | [2023.11.05-09.03.20:749][ 55]Full Startup: 67.72 seconds
asa_server2  | [2023.11.05-09.03.20:749][ 55]Number of cores 6
asa_server2  | [2023.11.05-09.03.25:217][ 56]wp.Runtime.HLOD = "1"
```

Could you send me the mod ids that you used? Might be a mod related issue.

**Comment by EvilSeeQ** [comment]
Do you mind sharing your .env file contents? (Please don't forget to edit out your passwords). I am suspecting you both are trying to launch the server with more than 1 mod, am i correct?

Seem to be something with the way it handles the mod list, anything more than 1 mod and it won't start (let it sit over night in case it was just slow to start with more than 1 mod, but nothing happened). Reducing it down to just 1 mod, and the server starts almost instantly.


Edit:

It was on my end, had a little typo in the .env file (not in the mod list though). Can just disregard the whole thing in my case.

Anyhow, heres the list of mods (just some of the most downloaded mods from curseforge i tried to test with):

```
# Mods: Please provide a comma seperated list of the curse-forge modIDs and uncomment the MODS arg
# e.g. MODS=931872,931338
MODS=900062,908148,912902,914844,916922,926956,927131,928597

```

**Comment by HylianSeven** [comment]
Not sure why it keeps coming out like this even though I use the "code" function.

```
# Basic server config

#APPID should only be changed if Wildcard provides a new one
ASA_APPID=2430930
SERVER_MAP=TheIsland_WP
SESSION_NAME="HylianSevenARK"
SERVER_PORT=7790
MAX_PLAYERS=10

# Comment to disable password
SERVER_PASSWORD=[redacted]
ARK_ADMIN_PASSWORD=[redacted]

# Comment to disable RCON
# If you disable RCON, the builtin manager won't work
RCON_PORT=32330

# Comment to enable BattlEye
#DISABLE_BATTLEYE=1

# Mods: Please provide a comma separated list of the curse-forge modIDs and uncomment the MODS arg
# e.g. MODS=931872,931338
# MODS=


# Extra arguments
ARK_EXTRA_OPTS="?ServerCrosshair=true?OverrideStructurePlatformPrevention=true?OverrideOfficialDifficulty=5.0?ShowFloatingDamageText=true?AllowFlyerCarryPvE=true"
ARK_EXTRA_DASH_OPTS="-ForceAllowCaveFlyers -ForceRespawnDinos -AllowRaidDinoFeeding=true -ActiveEvent=Summer"
```

Edit @azixus: Fixed formatting

**Comment by NIREKI** [comment]
@HylianSeven Could you please try to start the server with the default .env and let's say disable battleEye again?
Also, what are your server specs? Could you try monitoring during the container startup, and could you maybe even post the complete container output upon startup (e.g. docker container logs asa_server)

**Comment by HylianSeven** [comment]
I only changed the server name since there's another server already with the same default name, but here you go.

```


asa_server  | -------- STARTING SERVER --------
asa_server  | tail: /opt/arkserver/ShooterGame/Saved/Logs/ShooterGame.log: file                                                                                                                                                              truncated
asa_server  | [2023.11.05-15.30.05:054][  0]Log file open, 11/05/23 15:30:05
asa_server  | [2023.11.05-15.30.05:055][  0]LogMemory: Platform Memory Stats for                                                                                                                                                              WindowsServer
asa_server  | [2023.11.05-15.30.05:055][  0]LogMemory: Process Physical Memory:                                                                                                                                                              344.80 MB used, 346.97 MB peak
asa_server  | [2023.11.05-15.30.05:055][  0]LogMemory: Process Virtual Memory: 3                                                                                                                                                             16.07 MB used, 316.07 MB peak
asa_server  | [2023.11.05-15.30.05:055][  0]LogMemory: Physical Memory: 846.03 M                                                                                                                                                             B used,  7000.25 MB free, 7846.29 MB total
asa_server  | [2023.11.05-15.30.05:055][  0]LogMemory: Virtual Memory: 846.05 MB                                                                                                                                                              used,  7977.24 MB free, 8823.29 MB total
asa_server  | [2023.11.05-15.30.06:065][  0]ARK Version: 26.3
asa_server  | [2023.11.05-15.30.06:693][  0]Primal Game Data Took 0.44 seconds
asa_server  | [2023.11.05-15.30.07:108][  0]Server: "HylianSevenARK" has success                                                                                                                                                             fully started!
asa_server  | [2023.11.05-15.30.27:063][  0]Commandline:  TheIsland_WP?listen?Se                                                                                                                                                             ssionName="HylianSevenARK"?Port=7790?MaxPlayers=10?ServerPassword=[]?ServerAdminPassword="[]"?RCONEnabled=True?RCONPort=32330?ServerCro                                                                                                                                                             sshair=true?OverrideStructurePlatformPrevention=true?OverrideOfficialDifficulty=                                                                                                                                                             5.0?ShowFloatingDamageText=true?AllowFlyerCarryPvE=true -log -NoBattlEye -WinLiv                                                                                                                                                             eMaxPlayers=10 -ForceAllowCaveFlyers -ForceRespawnDinos -AllowRaidDinoFeeding=tr                                                                                                                                                             ue -ActiveEvent=Summer
asa_server  | [2023.11.05-15.30.27:063][  0]Full Startup: 24.22 seconds
asa_server  | [2023.11.05-15.30.27:063][  0]Number of cores 4
asa_server  | [2023.11.05-15.30.34:381][  1]wp.Runtime.HLOD = "1"
```

Edit @NIREKI: Removed Passwords

**Comment by NIREKI** [comment]
@HylianSeven I'm a bit confused but the server started completely according to the log output you've posted?

**Comment by HylianSeven** [comment]
That's what it looks like, but I'm not seeing anything in the sever browser. I just tried again a minute ago. Also I ran the status docker command and got this.

```
[redacted]@HylianARK:~/ASA_Server_Docker$ sudo docker compose exec asa_server manager status
[sudo] password for [redacted]:
Server PID not found (server offline?)
```

**Comment by azixus** [comment]
@HylianSeven you can dm me on Discord if you want, my username is the same as in GitHub. I might have a few ideas to debug. Will be easier to communicate.

**Comment by SBRyno** [comment]
> azixus

I have tried your suggestions and no change. I'll DM you on discord

**Comment by HylianSeven** [comment]
Thanks to azixus, we have figured out my issue lies between chair and keyboard: I did not have enough memory. I thought I did because I pulled out of of the sticks of RAM and saw "16GB" on it, but neglected to read the "2x8GB". I had to remove one stick of RAM because it seemingly died while the PC was sitting there doing nothing.

After speaking to SBRyno, that's not the same issue they are having, but at least we have an explanation for mine and I figured I should record it here.
